### ISSUE TITLE
Define language for scoping as derived model

### ISSUE THREAD

**Issue by enikao** [LionWeb-io/specification#177]
Calculating the scope of visible targets at any given place is necessary for both parser-based clients (for name binding / linking) and projectional clients / IDEs (for auto-completion).
Similar to validation, the basic structure of scoping is very similar in most systems; this _base language_ would condense the common concepts.

**Comment by dslmeinte** [comment]
_"base language"_ is something like KernelF: a funclarative language to compute "stuff" based on models?

**Comment by enikao** [comment]
> _"base language"_ is something like KernelF: a funclarative language to compute "stuff" based on models?

No, it's the "reporting language" we would pass to the API (#173).

A scoping example: a `DotExpression` node would have a derived node `Scope` with a list of references to all members of the left-hand side of the dot.

**Comment by dslmeinte** [comment]
Ah OK, so the language to report the result of the scope computation in the form of a derived model. I think we should (as with the review comments language) not strive to capture all contexts, but focus on providing a good and extensible default/base.
